docs/README-develop-best-practices.md

```diff
@@ -0,0 +1,385 @@
+# Develoment Best Practices for cisco_node_utils APIs.
+
+* [Overview](#overview)
+* [YAML Development Best Practices](#ydbp)
+* [Common Object Development Best Practices](#odbp)
+* [MiniTest Development Best Practices](#mdbp)
+
+## <a name="overview">Overview</a>
+
+This document is intended to assist in developing cisco_node_utils API's that are consistent with current best practices.
+
+
+## <a name="ydbp">YAML Development Best Practices</a>
```

Comment:
Be sure to update the section in README_YAML (https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/README_YAML.md#style-guide) to point to this document instead of having its own separate guidelines.